opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/drools/DroolsAlarmContext.java

```diff
@@ -180,19 +184,43 @@ public void handleDeletedAlarm(int alarmId, String reductionKey) {
         }
     }
 
+    private void handleRelatedAlarms(OnmsAlarm situation) {
+        if (!situation.isSituation()) {
+            return;
+        }
+        Map<Integer, AlarmAssociationAndFact> associationFacts = alarmAssociationById.get(situation.getId());
+        if (associationFacts == null) {
+            associationFacts = new HashMap<>();
+            alarmAssociationById.put(situation.getId(), associationFacts);
+        }
+        for (AlarmAssociation association : situation.getAssociatedAlarms()) {
```

Comment:
What happens if an associated alarm is removed? It seems that we won't actually remove the "alarm association fact" until the situation itself is deleted.